photon-ml/src/test/scala/com/linkedin/photon/ml/util/IdentityIndexMapLoaderTest.scala

```diff
@@ -0,0 +1,109 @@
+    // In unit test env, these two methods calls supposedly should return the same object
+    assertTrue(driverMap == rddMap)
+
+    assertEquals(driverMap.size, 10)
+
+    for (i <- 0 until 9) {
+      assertEquals(driverMap.get(String.valueOf(i)).get, i)
+      assertEquals(driverMap.getIndex(String.valueOf(i)), i)
+      assertEquals(driverMap.getFeatureName(i).get, String.valueOf(i))
+    }
+
+    assertEquals(driverMap.get(GLMSuite.INTERCEPT_NAME_TERM).get, 9)
+    assertEquals(driverMap.getIndex(GLMSuite.INTERCEPT_NAME_TERM), 9)
+    assertEquals(driverMap.getFeatureName(9).get, GLMSuite.INTERCEPT_NAME_TERM)
+
+    assertTrue(driverMap.get("RANDOM_KEY").isEmpty)
+    assertEquals(driverMap.getIndex("RANDOM_KEY"), IndexMap.NULL_KEY)
+    assertTrue(driverMap.getFeatureName(10).isEmpty)
+
+    val nameSet = mutable.Set[String]()
+    val idxSet = mutable.Set[Int]()
+    driverMap.map { case (name, idx) =>
+      if (idx != 9) {
+        assertEquals(name.toInt, idx)
+      } else {
+        assertEquals(name, GLMSuite.INTERCEPT_NAME_TERM)
+      }
+
+      nameSet.add(name)
+      idxSet.add(idx)
+    }
+
+    assertEquals(nameSet.size, 10)
+    assertEquals(idxSet.size, 10)
+    for (i <- 0 until 10) {
+      if (i != 9) {
+        assertTrue(nameSet.contains(i.toString()))
+      } else {
+        assertTrue(nameSet.contains(GLMSuite.INTERCEPT_NAME_TERM))
+      }
+      assertTrue(idxSet.contains(i))
+    }
+  }
+
+  @Test
+  def testMapWithoutIntercept(): Unit = {
+    val mapLoader = new IdentityIndexMapLoader(10, false)
+
+    assertFalse(mapLoader.useIntercept)
+    assertEquals(mapLoader.featureDimension, 10)
+
+    val driverMap = mapLoader.indexMapForDriver()
+    val rddMap = mapLoader.indexMapForRDD()
+    // In unit test env, these two methods calls supposedly should return the same object
+    assertTrue(driverMap == rddMap)
+
+    assertEquals(driverMap.size, 10)
+
+    for (i <- 0 until 10) {
+      assertEquals(driverMap.get(String.valueOf(i)).get, i)
+      assertEquals(driverMap.getIndex(String.valueOf(i)), i)
+      assertEquals(driverMap.getFeatureName(i).get, String.valueOf(i))
+    }
+
+    assertTrue(driverMap.get(GLMSuite.INTERCEPT_NAME_TERM).isEmpty)
+    assertEquals(driverMap.getIndex(GLMSuite.INTERCEPT_NAME_TERM), IndexMap.NULL_KEY)
+    assertEquals(driverMap.getFeatureName(9).get, "9")
+
+    assertTrue(driverMap.get("RANDOM_KEY").isEmpty)
+    assertEquals(driverMap.getIndex("RANDOM_KEY"), IndexMap.NULL_KEY)
+    assertTrue(driverMap.getFeatureName(10).isEmpty)
+
+    val nameSet = mutable.Set[String]()
+    val idxSet = mutable.Set[Int]()
+    driverMap.map { case (name, idx) =>
+      assertEquals(name.toInt, idx)
+      nameSet.add(name)
+      idxSet.add(idx)
+    }
+
+    assertEquals(nameSet.size, 10)
+    assertEquals(idxSet.size, 10)
+    for (i <- 0 until 10) {
+      assertTrue(nameSet.contains(i.toString()))
+      assertTrue(idxSet.contains(i))
+    }
+  }
+}
```

Comment:
Please add a newline after the last bracket. Also, LGTM, squash the PR into one commit (the changes are small enough that I think it would be best that way) and I'll merge